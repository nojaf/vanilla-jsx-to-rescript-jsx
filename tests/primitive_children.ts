import { test, expect } from "bun:test";
import transfrom from "../index";

test("13 became React.int", () => {
  expect(transfrom("<span>13</span>")).toMatchSnapshot();
});

test(" 7  became React.int", () => {
  expect(transfrom("<span> 7 </span>")).toMatchSnapshot();
});

test("4.2 became React.float", () => {
  expect(transfrom("<span>4.2</span>")).toMatchSnapshot();
});

test("33.7 became React.float", () => {
  expect(transfrom("<span> 33.7 </span>")).toMatchSnapshot();
});

test("single space does not become React.string", () => {
  expect(transfrom("<span> </span>")).toMatchSnapshot();
});

test("newline does not become React.string", () => {
  expect(
    transfrom(`<span>
</span>`),
  ).toMatchSnapshot();
});
