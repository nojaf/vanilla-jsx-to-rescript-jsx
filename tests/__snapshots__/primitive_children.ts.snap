// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`13 became React.int 1`] = `"<span>{React.int(13)}</span>"`;

exports[` 7  became React.int 1`] = `"<span>{React.int( 7 )}</span>"`;

exports[`4.2 became React.float 1`] = `"<span>{React.float(4.2)}</span>"`;

exports[`33.7 became React.float 1`] = `"<span>{React.float( 33.7 )}</span>"`;

exports[`single space does not become React.string 1`] = `"<span> </span>"`;

exports[`newline does not become React.string 1`] = `
"<span>
</span>"
`;
